db/1.8/modules/ROOT/pages/database-connector-udt-stored-procedure.adoc

```diff
@@ -0,0 +1,328 @@
++
+[%header,cols="30a,70a"]
+|===
+| Id | Type name
+| 2003 |  `PEOPLE`
+| 2003 |  `PHONE_NUMBER`
+| 2008 |  `PERSON`
+| 2003 |  `PHONE_ARRAY`
+| 2003 |  `PHONE_BOOK`
+|===
++
+[start=13]
+. Click *Finish* to close the *Column type* window.
+. Click *Test Connection* to confirm that Mule can connect to the database.
+. Click *OK*.
+
+The following screenshot shows the HTTP Listener global element configuration in Studio:
+
+.HTTP Listener configuration
+image::database-oracleudt-example-1.png[HTTP Listener configuration]
+
+The following screenshots shows the Database global element configuration in Studio:
+
+.Database Config General cofiguration
+image::database-oracleudt-example-2.png[Database Config General configuration setting Host, Port, User, Password and Instance values parameters]
+
+.Database Config Advanced cofiguration
+image::database-oracleudt-example-3.png[Database Config advanced configuration setting Column types values]
+
+
+== Create and Run for your Mule App
+
+Create the Mule app flow in Anypoint Studio:
+
+. In the *Mule Palette* view, select the HTTP *Listener* source and drag it on to the canvas. +
+The source listens for incoming HTTP messages attributes.
+. In the *Connector configuration* field, select `HTTP_Listener_config` global configuration.
+. Set the *Path* field to `/oracle`.
+. Drag a *Set Payload* component to the right of *Listener*. +
+This component set a list of items to send to the stored procedure to process.
+. In the *Value* field, set the following code for the list of items: +
++
+[source,xml,linenums]
+----
+[{'name':'Anthony J', 'surname':'Crowley', 'age': 6000, 'phoneNumber': {'areaCode':'020', 'phoneNumber': '777'}},
+	{'name':'A.Z', 'surname':'Fell', 'age': 6000, 'phoneNumber':{'areaCode':'020', 'phoneNumber': '888'}},
+	{'name':'Adam', 'surname':'Young', 'age': 11, 'phoneNumber':{'areaCode':'046', 'phoneNumber': '666'}},
+	{'name':'Anathema', 'surname':'Device', 'age': 27, 'phoneNumber':{'areaCode':'020', 'phoneNumber': '123'}},
+	]
+----
++
+
+.Set Payload configuration
+image::database-oracleudt-example-4.png[Set payload configuration]
+
+[start=7]
+. Set the *Mime Type* field to `application/java`
+. Drag a *Transform Message* component to the right of *Set Payload*. +
+This component transforms the JSON objects into an object that can be mapped to the data type `PEOPLE` that the stored procedure expects as its input parameter.
+. Double click the component in the Studio canvas, and set the name to `Transform Message - Prepare UDT`.
+. In the *Output* source code view of the component, click the *Edit Current Target* button.
+. In the *Selection dialog* box, set the *Output* field to `Variable`, the *Variable name* to `in_people_tab`, and click *OK*.
+. In the *Output* field, set the following DataWeave expression that contains the `createStruct`  and `createArray` functions to create said object:
```

Comment:
```suggestion
. In the *Output* field, specify the following DataWeave expression:
```
I would omit the modifiers about the content and its purpose: it is stated earlier, contains an extra space, and it is unclear to what "said object" refers.